### PR TITLE
Prevent leaking credentials (VIALJS-85)

### PR DESCRIPTION
**Pull request by c0deaddict**
## Purpose

Already committed credentials in the example file by accident already once. Test for empty values for store keys, so we don't need to reset credentials next time.

## Approach

Also, add git guppy (https://github.com/therealklanni/git-guppy) to run the gulp test task on pre-commit.

#### Open Questions and Pre-Merge TODOs
- [ ] Use github checklists. When solved, check the box and explain the answer.

## Learning

Also added eslint and the whole of test-unit to the pre-commit hook.

Depends on #189 and #188

### CODE REVIEW

File: tools/helpers.js [hunk 1]
```diff
@@ -500,6 +502,53 @@ class Helpers {
...
+            'clientSecret',
+            'refreshToken',
+            'username',
+            'password',
+        ]
+
+        function scanForProtectedKeys(file, obj, attrPath) {
+            let count = 0
+            Object.entries(obj).forEach(([key, value]) => {
+                if (protectedKeys.indexOf(key) !== -1 && value) {
```
Comment (ghost) [inline tools/helpers.js]: `.includes` is a bit easier to read imo

File: tools/helpers.js [hunk 2]
```diff
@@ -33,6 +33,8 @@ const source = require('vinyl-source-stream')
 const sourcemaps = require('gulp-sourcemaps')
 const through = require('through2')
 const watchify = require('watchify')
+const map = require('map-stream')
```
Comment (ghost) [inline tools/helpers.js:36]: Add to package.json?

File: gulpfile.js [hunk 3]
```diff
@@ -504,5 +545,8 @@ gulp.task('watch', 'Run developer watch modus.', () => {
         path.join(settings.NODE_PATH, 'vjs-mod-*', 'src', 'components', '**', '*.vue'),
     ], ['templates'])
 
-    gulp.watch(path.join(settings.ROOT_DIR, 'test', 'bg', '**', '*.js'), ['test-unit'])
+    gulp.watch([
+        path.join(settings.ROOT_DIR, 'test', 'bg', '**', '*.js'),
+        path.join(settings.SRC_DIR, '**', '*.js'),
```
Comment (ghost) [inline gulpfile.js]: At the moment there is a WATCHTEST flag that determines whether to run unit-tests after a build. Doesn't this interfere with `gulp watch --verify`?

**Review by ghost — CHANGES_REQUESTED** [review]

File: tools/helpers.js (hunk 1 above)
Comment (c0deaddict) [reply tools/helpers.js]: Nice, didn't know about that method :+1:

File: gulpfile.js (hunk 3 above)
Comment (c0deaddict) [reply gulpfile.js]: Good point, hadn't seen that flag. Now the tests are running twice, with `--verify`, so i'll remove the watch on `SRC_DIR`.

**Pull request merged**